Fix git-rpm-version behaviour with semver2.0.0 Release Candidate numbering

endemics opened the pull request:
Semver 2.0.0 has changed the way Release Candidates are numbered, and introduced a '.' between 'rc' and the number.

This PR aims at making `git-rpm-version` compatible with this new scheme.

endemics committed:
Adding tests for Semver 2.0.0 release candidates, as it has introduced a "." between rc and the number

endemics committed:
Fixing code to pass semver 2.0.0 release candidate tests

peterlundgren commented:
From the readme:

```
Tree                         RPM Name
---------------------------  --------------------------
...
* 9dbedc2 (tag: v2.0.0-rc1)  foo-2.0.0-0.rc1.rpm
...
```

If I understand correctly, your patch implements:

```
Tree                          RPM Name
----------------------------  --------------------------
...
* 9dbedc2 (tag: v2.0.0-rc.1)  foo-2.0.0-0.rc1.rpm
...
```

We could also change the RPM release to match the semver 2 convention like this:

```
Tree                          RPM Name
----------------------------  --------------------------
...
* 9dbedc2 (tag: v2.0.0-rc.1)  foo-2.0.0-0.rc.1.rpm
...
```

What do you think about that?

endemics committed:
It turns out that vercmp seems to treat "rc.X" as equal to "rcX",
so no need to mangle release candidates versions respecting semver 2.0.0
and forcing them back to semver 1.0.0 compatibility in the RPM name.

endemics commented:
Hi

Yes you are correct in your interpretation, in that I transformed the semver 2.0.0 rc tags "back" to semver 1.0.0 versions. The reason was mainly that I was worried with potential side effects in comparisons and it like a felt more "conservative" approach:  I knew how the rcX RPM versions were going to behave at update time, and wasn't too sure how the rc.X RPM versions would behave.

Since I have checked some examples with `rpmdev-vercmp` and it basically seems to consider them identical so I guess it should be safe to respect the semver 2.0.0 notation.

For instance, on 2 git commits after a rc1 tag we would have:
`rpmdev-vercmp foo-2.0.0-0.rc1.2.g0453d18.el7.x86_64 foo-2.0.0-0.rc.1.2.g0453d18.el7.x86_64
foo-2.0.0-0.rc1.2.g0453d18.el7.x86_64 == foo-2.0.0-0.rc.1.2.g0453d18.el7.x86_64`

Will update the PR.
